[BUG] v3.0.7: libsodium fails to build

### Database name

n/a

### WAL-G Version

v3.0.7

### Describe your problem

I build wal-g with libsodium in a Dockerfile. As of v3.0.7 (the issue doesn't happen with v3.0.6), the `GOBIN=/go/bin USE_LIBSODIUM=1 USE_LZO=1 make install_and_build_pg` fails. 

### Steps to reproduce

1. Run `docker buildx build --push . --build-arg BASE=17.4 --build-arg WALG_RELEASE=v3.0.7 --ulimit memlock=16384:16384 --platform linux/amd64` with [this Dockerfile](https://git.koehn.com/docker/postgres-wal-g/raw/branch/master/Dockerfile)
2. Get error below. 

```
(cd main/pg && go build -mod vendor -tags " libsodium lzo" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/pg.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/pg.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/pg.walgVersion=`git tag -l --points-at HEAD`")
# github.com/wal-g/wal-g/main/pg
usage: link [options] main.o
  -B note
    	set ELF NT_GNU_BUILD_ID note or Mach-O UUID; use "gobuildid" to generate it from the Go build ID; "none" to disable
  -E entry
    	set entry symbol name
  -H type
    	set header type
```

Not sure if it's the leading space in `" libsodium lzo"` or something else. 

### Config and WAL-G logs

```
139.2  /usr/bin/install -c -m 644 libsodium.pc '/go/src/wal-g/tmp/libsodium/lib/pkgconfig'
139.2 make[3]: Leaving directory '/go/src/wal-g/tmp/libsodium'
139.2 make[2]: Leaving directory '/go/src/wal-g/tmp/libsodium'
139.2 make[1]: Leaving directory '/go/src/wal-g/tmp/libsodium'
139.4 (cd main/pg && go build -mod vendor -tags " libsodium lzo" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/pg.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/pg.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/pg.walgVersion=`git tag -l --points-at HEAD`")
159.4 # github.com/wal-g/wal-g/main/pg
159.4 usage: link [options] main.o
159.4   -B note
159.4     	set ELF NT_GNU_BUILD_ID note or Mach-O UUID; use "gobuildid" to generate it from the Go build ID; "none" to disable
159.4   -E entry
159.4     	set entry symbol name
159.4   -H type
159.4     	set header type
159.4   -I linker
159.4     	use linker as ELF dynamic linker
159.4   -L directory
159.4     	add specified directory to library path
159.4   -R quantum
159.4     	set address rounding quantum (default -1)
159.4   -T int
159.4     	set the start address of text symbols (default -1)
159.4   -V	print version and exit
159.4   -X definition
159.4     	add string value definition of the form importpath.name=value
159.4   -a	no-op (deprecated)
159.4   -asan
159.4     	enable ASan interface
159.4   -aslr
159.4     	enable ASLR for buildmode=c-shared on windows (default true)
159.4   -benchmark string
159.4     	set to 'mem' or 'cpu' to enable phase benchmarking
159.4   -benchmarkprofile base
159.4     	emit phase profiles to base_phase.{cpu,mem}prof
159.4   -bindnow
159.4     	mark a dynamically linked ELF object for immediate function binding
159.4   -buildid id
159.4     	record id as Go toolchain build id
159.4   -buildmode mode
159.4     	set build mode
159.4   -c	dump call graph
159.4   -capturehostobjs string
159.4     	capture host object files loaded during internal linking to specified dir
159.4   -checklinkname
159.4     	check linkname symbol references (default true)
159.4   -compressdwarf
159.4     	compress DWARF if possible (default true)
159.4   -cpuprofile file
159.4     	write cpu profile to file
159.4   -d	disable dynamic executable
159.4   -debugnosplit
159.4     	dump nosplit call graph
159.4   -debugtextsize int
159.4     	debug text section max size
159.4   -debugtramp int
159.4     	debug trampolines
159.4   -dumpdep
159.4     	dump symbol dependency graph
159.4   -extar string
159.4     	archive program for buildmode=c-archive
159.4   -extld linker
159.4     	use linker when linking in external mode
159.4   -extldflags flags
159.4     	pass flags to external linker
159.4   -f	ignore version mismatch
159.4   -fipso file
159.4     	write fips module to file
159.4   -g	disable go package data checks
159.4   -h	halt on error
159.4   -importcfg file
159.4     	read import configuration from file
159.4   -installsuffix suffix
159.4     	set package directory suffix
159.4   -k symbol
159.4     	set field tracking symbol
159.4   -libgcc string
159.4     	compiler support lib for internal linking; use "none" to disable
159.4   -linkmode mode
159.4     	set link mode
159.4   -linkshared
159.4     	link against installed Go shared libraries
159.4   -memprofile file
159.4     	write memory profile to file
159.4   -memprofilerate rate
159.4     	set runtime.MemProfileRate to rate
159.4   -msan
159.4     	enable MSan interface
159.4   -n	no-op (deprecated)
159.4   -o file
159.4     	write output to file
159.4   -pluginpath string
159.4     	full path name for plugin
159.4   -pruneweakmap
159.4     	prune weak mapinit refs (default true)
159.4   -r path
159.4     	set the ELF dynamic linker search path to dir1:dir2:...
159.4   -race
159.4     	enable race detector
159.4   -randlayout int
159.4     	randomize function layout
159.4   -s	disable symbol table
159.4   -strictdups int
159.4     	sanity check duplicate symbol contents during object file reading (1=warn 2=err).
159.4   -tmpdir directory
159.4     	use directory for temporary files
159.4   -v	print link trace
159.4   -w	disable DWARF generation
159.5 make: *** [Makefile:50: pg_build] Error 1
------
Dockerfile:7
--------------------
   6 |
   7 | >>> RUN apt-get update && apt-get install -qqy --no-install-recommends --no-install-suggests cmake gnutls-bin liblzo2-dev libsodium23 libsodium-dev && \
   8 | >>>     cd /go/src/ && \
   9 | >>>     git clone -b $WALG_RELEASE --recurse-submodules https://github.com/wal-g/wal-g.git && \
  10 | >>>     cd wal-g && \
  11 | >>>     GOBIN=/go/bin go mod tidy && \
  12 | >>>     GOBIN=/go/bin USE_LIBSODIUM=1 USE_LZO=1 make install_and_build_pg
  13 |
--------------------
ERROR: failed to solve: process "/bin/sh -c apt-get update && apt-get install -qqy --no-install-recommends --no-install-suggests cmake gnutls-bin liblzo2-dev libsodium23 libsodium-dev &&     cd /go/src/ &&     git clone -b $WALG_RELEASE --recurse-submodules https://github.com/wal-g/wal-g.git &&     cd wal-g &&     GOBIN=/go/bin go mod tidy &&     GOBIN=/go/bin USE_LIBSODIUM=1 USE_LZO=1 make install_and_build_pg" did not complete successfully: exit code: 2
```

--- a/Makefile
+++ b/Makefile
@@ -25,6 +25,8 @@ IMAGE_TYPE ?= "rdb"
 TOOLS_MOD_DIR := ./internal/tools
 MOCKS_DESTINATION := ./testtools/mocks
 FILE_TO_MOCKS := ./internal/uploader.go # list interface paths here
+WALG_VERSION ?= `git tag -l --points-at HEAD | tail -1`
+GIT_REVISION ?= `git rev-parse --short HEAD`
 
 BUILD_TAGS:=
 
@@ -47,7 +49,7 @@ test: deps unittest pg_build mysql_build redis_build mongo_build gp_build cloudb
 pg_test: deps pg_build unlink_brotli pg_integration_test
 
 pg_build: $(CMD_FILES) $(PKG_FILES)
-	(cd $(MAIN_PG_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/pg.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/pg.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/pg.walgVersion=`git tag -l --points-at HEAD`")
+	(cd $(MAIN_PG_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/pg.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/pg.gitRevision=$(GIT_REVISION) -X github.com/wal-g/wal-g/cmd/pg.walgVersion=$(WALG_VERSION)")
 
 install_and_build_pg: deps pg_build
 
@@ -124,10 +126,10 @@ mysql_base: deps mysql_build unlink_brotli
 mysql_test: deps mysql_build unlink_brotli mysql_integration_test
 
 mysql_build: $(CMD_FILES) $(PKG_FILES)
-	(cd $(MAIN_MYSQL_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/mysql.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/mysql.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/mysql.walgVersion=`git tag -l --points-at HEAD`")
+	(cd $(MAIN_MYSQL_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/mysql.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/mysql.gitRevision=$(GIT_REVISION) -X github.com/wal-g/wal-g/cmd/mysql.walgVersion=$(WALG_VERSION)")
 
 sqlserver_build: $(CMD_FILES) $(PKG_FILES)
-	(cd $(MAIN_SQLSERVER_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/sqlserver.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/sqlserver.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/sqlserver.walgVersion=`git tag -l --points-at HEAD`")
+	(cd $(MAIN_SQLSERVER_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/sqlserver.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/sqlserver.gitRevision=$(GIT_REVISION) -X github.com/wal-g/wal-g/cmd/sqlserver.walgVersion=$(WALG_VERSION)")
 
 load_docker_common:
 	@if [ "x" = "${CACHE_FOLDER}x" ]; then\
@@ -165,7 +167,7 @@ mariadb_integration_test: unlink_brotli load_docker_common
 mongo_test: deps mongo_build unlink_brotli
 
 mongo_build: $(CMD_FILES) $(PKG_FILES)
-	(cd $(MAIN_MONGO_PATH) && go build $(BUILD_ARGS) -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/mongo.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/mongo.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/mongo.walgVersion=`git tag -l --points-at HEAD`")
+	(cd $(MAIN_MONGO_PATH) && go build $(BUILD_ARGS) -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/mongo.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/mongo.gitRevision=$(GIT_REVISION) -X github.com/wal-g/wal-g/cmd/mongo.walgVersion=$(WALG_VERSION)")
 
 mongo_install: mongo_build
 	mv $(MAIN_MONGO_PATH)/wal-g $(GOBIN)/wal-g
@@ -193,7 +195,7 @@ fdb_integration_test: load_docker_common
 redis_test: deps redis_build unlink_brotli redis_integration_test
 
 redis_build: $(CMD_FILES) $(PKG_FILES)
-	(cd $(MAIN_REDIS_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/redis.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/redis.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/redis.walgVersion=`git tag -l --points-at HEAD`")
+	(cd $(MAIN_REDIS_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/redis.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/redis.gitRevision=$(GIT_REVISION) -X github.com/wal-g/wal-g/cmd/redis.walgVersion=$(WALG_VERSION)")
 
 redis_integration_test: load_docker_common
 	docker compose build redis && docker compose build redis_tests
@@ -218,7 +220,7 @@ clean_redis_features:
 etcd_test: deps etcd_build unlink_brotli etcd_integration_test
 
 etcd_build: $(CMD_FILES) $(PKG_FILES)
-	(cd $(MAIN_ETCD_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/etcd.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/etcd.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/etcd.walgVersion=`git tag -l --points-at HEAD`")
+	(cd $(MAIN_ETCD_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/etcd.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/etcd.gitRevision=$(GIT_REVISION) -X github.com/wal-g/wal-g/cmd/etcd.walgVersion=$(WALG_VERSION)")
 
 etcd_install: etcd_build
 	mv $(MAIN_ETCD_PATH)/wal-g $(GOBIN)/wal-g
@@ -233,7 +235,7 @@ etcd_integration_test: load_docker_common
 	docker compose up --exit-code-from etcd_tests etcd_tests
 
 gp_build: $(CMD_FILES) $(PKG_FILES)
-	(cd $(MAIN_GP_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/gp.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/gp.gitRevision=`git rev-parse --short HEAD` -X github.com/wal-g/wal-g/cmd/gp.walgVersion=`git tag -l --points-at HEAD`")
+	(cd $(MAIN_GP_PATH) && go build -mod vendor -tags "$(BUILD_TAGS)" -o wal-g -ldflags "-s -w -X github.com/wal-g/wal-g/cmd/gp.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X github.com/wal-g/wal-g/cmd/gp.gitRevision=$(GIT_REVISION) -X github.com/wal-g/wal-g/cmd/gp.walgVersion=$(WALG_VERSION)")
 
 gp_clean:
 	(cd $(MAIN_GP_PATH) && go clean)
@@ -335,7 +337,7 @@ unlink_libsodium:
 
 build_client:
 	cd cmd/daemonclient && \
-	go build -o ../../bin/walg-daemon-client -ldflags "-s -w -X main.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X main.gitRevision=`git rev-parse --short HEAD` -X main.version=`git tag -l --points-at HEAD`"
+	go build -o ../../bin/walg-daemon-client -ldflags "-s -w -X main.buildDate=`date -u +%Y.%m.%d_%H:%M:%S` -X main.gitRevision=$(GIT_REVISION) -X main.version=$(WALG_VERSION)"
 
 .PHONY: mocks
 # put the files with interfaces you'd like to mock in prerequisites

